### ISSUE TITLE
isPacket elementary routine into tfe?

### ISSUE THREAD

**Issue by DavidBrainard** [gkaguirrelab/temporalFittingEngine#10]
I looked at the isPacket routine in TPUP.  Except possibly for the very last check, it appears generic in that we'd want those checks for any model.  I suggest we move this into the parent class as a method.  Specific models can then call this in their own isPacket and if it passes continue on to do model specific checks.  If there are no model specific checks, it would not be necessary to implement a separate isPacket in the subclass.  This is one of the beauties of object oriented programming, which makes it easy to reuse generic code but then override or extend without changing the calling method syntax.  If there is agreement that there is a good concept, I can implement.  I'll need to grab Nicolas or Ben to show me the right syntax, but then we will have an example.

**Comment by gkaguirre** [comment]
This sounds like a win. I'll be interested to see how the subclass isPacket adds on additional checks to the those that are standard in the top-level the class.

**Comment by DavidBrainard** [comment]
I implemented this in a branch, created a pull request, and pulled it into a master.  What I did was put the basic isPacket method into the tfe parent class.  Then in the TPUPModel, the isPacket first calls the parent method and then checks that the first response value is near zero.  (The parent class method omits this last check, which I deemed to be model specific).

I did not test this, because I didn't see a clear test program to do it with.  But when we have a test program that exercises isPacket, we will find out whether this works.  In principle, models without an explicit isPacket should be able to call the method and will obtain the generic behavior.
